ArcGISRuntimeSDKQt_CppSamples/Analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.cpp

```diff
@@ -0,0 +1,368 @@
+    case UtilityAttributeComparisonOperator::DoesNotIncludeTheValues:
+      return QString("DoesNotIncludeTheValues");
+    case UtilityAttributeComparisonOperator::IncludesAny:
+      return QString("IncludesAny");
+    case UtilityAttributeComparisonOperator::DoesNotIncludeAny:
+      return QString("DoesNotIncludeAny");
+  }
+}
+
+QVariant ConfigureSubnetworkTrace::convertToDataType(const QVariant& value, const Esri::ArcGISRuntime::UtilityNetworkAttributeDataType& dataType)
+{
+  switch (dataType)
+  {
+    case UtilityNetworkAttributeDataType::Integer:
+    {
+      return static_cast<int>(value.toDouble());
+    }
+    case UtilityNetworkAttributeDataType::Float:
+    {
+      return value.toFloat();
+    }
+    case UtilityNetworkAttributeDataType::Double:
+    {
+      return value.toDouble();
+    }
+    case UtilityNetworkAttributeDataType::Boolean:
+    {
+      return value.toBool();
+    }
+  }
+}
+
+void ConfigureSubnetworkTrace::codedValueOrInputText(const QString& currentText)
+{
+  // Update the UI to show the correct value entry for the attribute.
+  if (m_networkDefinition)
+  {
+    const Domain domain = m_networkDefinition->networkAttribute(currentText)->domain();
+    if (!domain.isEmpty() && (domain.domainType() == DomainType::CodedValueDomain))
+    {
+      m_valueSelectionListModel.clear();
+      const CodedValueDomain codedValueDomain = static_cast<CodedValueDomain>(domain);
+
+      for (CodedValue codedValue: codedValueDomain.codedValues())
+        m_valueSelectionListModel.append(codedValue.name());
+
+      m_textFieldVisible = false;
+    }
+    else
+    {
+      m_textFieldVisible = true;
+    }
+    emit valueSelectionListModelChanged();
+    emit textFieldVisibleChanged();
+  }
+}
+
+void ConfigureSubnetworkTrace::addCondition(const QString& selectedAttribute, int selectedOperator, const QVariant& selectedValue)
+{
+  // NOTE: You may also create a UtilityCategoryComparison with UtilityNetworkDefinition.Categories and UtilityCategoryComparisonOperator.
+
+  if (!m_traceConfiguration)
+    m_traceConfiguration = new UtilityTraceConfiguration(this);
+
+  UtilityTraversability* traversability = m_traceConfiguration->traversability();
+  if (!traversability)
+    m_traceConfiguration->setTraversability(new UtilityTraversability(this));
+
+  UtilityNetworkAttribute* selectedNetworkAttribute = m_networkDefinition->networkAttribute(selectedAttribute);
+
+  const QVariant convertedSelectedValue = convertToDataType(selectedValue, selectedNetworkAttribute->dataType());
+
+  const UtilityAttributeComparisonOperator selectedOperatorEnum = static_cast<UtilityAttributeComparisonOperator>(selectedOperator);
+
+
+  // NOTE: You may also create a UtilityNetworkAttributeComparison with another NetworkAttribute.
+  UtilityTraceConditionalExpression* expression = new UtilityNetworkAttributeComparison(selectedNetworkAttribute, selectedOperatorEnum, convertedSelectedValue, this);
+
+  if (traversability->barriers())
```

Comment:
This could be a problem. Above, you _might_ execute `m_traceConfiguration->setTraversability(new UtilityTraversability(this));`, so this could be `nullptr` in that case.